### PR TITLE
Eve-7 Use unique_ptr for TAxis in REveCaloData

### PR DESCRIPTION
**Pull request by alja**

### CODE REVIEW

**Comment by alja** [comment]
@linev Please check.

File: graf3d/eve7/src/REveCaloData.cxx [hunk 1]
```diff
@@ -818,6 +808,10 @@ void REveCaloDataHist::GetCellData(const REveCaloData::CellId_t &id,
 
 Int_t REveCaloDataHist::AddHistogram(TH2F* hist)
 {
+    if (!fEtaAxis) {
+      fEtaAxis.reset(hist->GetXaxis());
+      fPhiAxis.reset(hist->GetYaxis());
+   }
```
Comment (linev) [inline graf3d/eve7/src/REveCaloData.cxx:814]: This is wrong. Assign pointer to unique_ptr one also delegates ownership.
Here one have to create copy of TAxis

File: graf3d/eve7/src/REveCaloData.cxx [hunk 2]
```diff
@@ -636,11 +631,8 @@ void  REveCaloDataVec::SetAxisFromBins(Double_t epsX, Double_t epsY)
    }
    newY.push_back(binY.back()); // overflow
 
-   if (fEtaAxis) delete fEtaAxis;
-   if (fPhiAxis) delete fPhiAxis;
-
-   fEtaAxis = new TAxis(newX.size()-1, &newX[0]);
-   fPhiAxis = new TAxis(newY.size()-1, &newY[0]);
+   fEtaAxis.reset(new TAxis(newX.size()-1, &newX[0]));
+   fPhiAxis.reset(new TAxis(newY.size()-1, &newY[0]));
```
Comment (linev) [inline graf3d/eve7/src/REveCaloData.cxx:635]: Here better to use:
```
fEtaAxis = std::make_unique<TAxis>(newX.size()-1, &newX[0])
...
```

File: graf3d/eve7/src/REveCaloData.cxx (hunk 1 above)
Comment (linev) [reply graf3d/eve7/src/REveCaloData.cxx:814]: @osschar, same problem is in EVE,

**Comment by alja** [comment]
@linev I have applied the changes.

**Comment by linev** [comment]
@phsft-bot build

**Review by linev — APPROVED** [review]
Now the ownership should be right.

**Pull request merged**